src/runtime/pythonengine.cs

```diff
@@ -64,7 +67,17 @@ public static string ProgramName
 
                 return result ?? "";
             }
-            set { Runtime.Py_SetProgramName(value); }
+            set
+            {
+                if (_programName != IntPtr.Zero)
+                {
+                    Marshal.FreeHGlobal(_programName);
+                }
+                _programName = Runtime.IsPython3
+                    ? UcsMarshaler.GetInstance("").MarshalManagedToNative(value)
```

Comment:
it is already. https://github.com/pythonnet/pythonnet/blob/0a478024175e52106d9d4ff2230e1f505e56d7d8/src/runtime/CustomMarshaler.cs#L166